server/db.py

```diff
@@ -192,6 +196,35 @@ def clean_db(self):
             self.clear_excess_history(self.utxo_flush_count)
         self.clear_excess_undo_info()
 
+    def fs_update_header_offsets(self, offset_start, height_start, headers):
+        if self.coin.STATIC_BLOCK_HEADERS:
+            return
+        # Write dynamic header offsets, each offset is 64 bits
+        offset = offset_start
+        offsets = []
+        for h in headers:
+            offset += len(h)
+            offsets.append(pack("<Q", offset))
+        self.headers_offsets_file.write(height_start * 8, b''.join(offsets))
+
+    def dynamic_header_offset(self, height):
+        assert not self.coin.STATIC_BLOCK_HEADERS
+        # The offsets are 64 bits in order to accommodate coins with big
+        # headers that will accumulate GBs of header data after some years
+        # We don't store the offset of the genesis header as it's always 0
+        # so this means that the offset for height 1 is stored at
+        # position 0, for height 2 at 8 and so on.
+        if height > 0:
```

Comment:
Yes, block 0 offset is a special case to "bootstrap" the first offset when the `headers_offsets_file` is initially empty, otherwise `dynamic_header_offset` will not work. 

Check how the `fs_update_header_offsets` calculates the offset of block N based on the header size of block N-1.

One pythonic "fix" could be:

```
try:
    pos = (height - 1) * 8
    offset, = unpack('<Q', self.headers_offsets_file.read(pos, 8))
    return offset
except struct.error as e:
    if height == 0:
        return 0
    else:
        raise Exception('Header offset not found: {}'.format(height))
```

Another alternative is to check and write the block 0 offset when creating the `headers_offsets_file`.